RTNeural/conv1d/conv1d.h

```diff
@@ -190,10 +195,47 @@ class Conv1DT
         state_ptr = (state_ptr == state_size - 1 ? 0 : state_ptr + 1); // iterate state pointer forwards
     }
 
+    template<int _groups_of = groups_of, std::enable_if_t<_groups_of != 1, bool> = true>
+    /** Performs forward propagation for this layer. */
+    inline void forward(const T (&ins)[in_size]) noexcept
+    {
+        // insert input into a circular buffer
+        std::copy(std::begin(ins), std::end(ins), state[state_ptr].begin());
+
+        // set state pointers to particular columns of the buffer
+        setStatePointers();
+
+        // perform multi-channel convolution
+        for(int i = 0; i < out_size; ++i)
+        {
+            outs[i] = bias[i];
+
+            // copy selected columns to a helper variable
+            for(int k = 0; k < kernel_size; ++k)
+            {
+                const auto& column = state[state_ptrs[k]];
+                const auto column_begin = column.begin() + i * group_count;
+                const auto column_end = column.begin() + i * group_count + group_count;
+                std::copy(column_begin, column_end, state_cols[k].begin());
+            }
+
+            for(int k = 0; k < kernel_size; ++k)
+            {
+                outs[i] = std::inner_product(
+                    weights[i][k].begin(),
+                    weights[i][k].end(),
+                    state_cols[k].begin(),
+                    outs[i]);
+            }
```

Comment:
This is cool! Do you think it would make sense to merge these two loops, or maybe there's some reason the logic should be kept separate? I don't think it would make any difference from a performance perspective, so I'm more just thinking about what would make the most sense to a potential reader.

Reply:
I separated them initially to make it close to `groups_of=1` (and also for debugging purposes). I think merging them is fine now.